src/internal/m365/controller.go

```diff
@@ -75,6 +74,17 @@ func NewController(
 		return nil, clues.Wrap(err, "creating api client").WithClues(ctx)
 	}
 
+	rc := resource.UnknownResource
```

Comment:
This is the heart of the fix.  All other changes are deletions to comply with moving the rc identification into this func.